### PR TITLE
Language and formatting fixes as a follow up to #1956

### PR DESCRIPTION
**Pull request by rkratky**
This PR fixes some minor language a formatting issues that were mostly introduced with #1956.

(Aside from punctuation and other small things, I removed hard line breaks to make it consistent with the rest of the doc.

### CODE REVIEW

File: doc/reference/autoinstall-reference.rst [hunk 1]
```diff
@@ -18,25 +15,13 @@ Here is an example of a minimal autoinstall configuration:
...
 
 .. note::
 
-  Technically, in all but one case the top level ``autoinstall`` keyword is
-  strictly unnecessary. This keyword is only necessary when serving autoinstall
-  via cloud-config. For backwards compatibility this format is still supported
-  for non-cloud-config based delivery methods; however, it is
-  **highly recommended** to use the format with a top-level ``autoinstall``
-  keyword as mistakes in this formatting are a common source of confusion.
+    This behaviour was first introduced during 24.04 (Noble). On any ISOs built before 24.04, you need to refresh the installer to see this behaviour.
```
Comment (dbungert) [inline doc/reference/autoinstall-reference.rst:22]: I think I want to refer back to https://canonical-subiquity--1963.com.readthedocs.build/en/1963/intro-to-autoinstall.html#providing-the-autoinstall-configuration to help explain the pre-24.04 behavior.  Would you add some text to that end please?

Comment (Chris-Peterson444) [reply doc/reference/autoinstall-reference.rst:22]: Agreed. I left a comment in #1962 about making this difference clear there, so once that's done I think we could just point to that.

**Review by Chris-Peterson444 — APPROVED** [review]
Other than the note about the reference to the other section this LGTM. 

Curious: I noticed a few of the changes were mostly moving the text back to one line (I think). Is this intentional? It's fine either way, just curious if it had to do with how the page is built or anything like that.

File: doc/reference/autoinstall-reference.rst (hunk 1 above)
Comment (rkratky) [reply doc/reference/autoinstall-reference.rst:22]: I updated #1962 as suggested.

**Comment by rkratky** [comment]
> Other than the note about the reference to the other section this LGTM.
> 
> Curious: I noticed a few of the changes were mostly moving the text back to one line (I think). Is this intentional? It's fine either way, just curious if it had to do with how the page is built or anything like that.

Yes, I noted it in the PR description:

_Aside from punctuation and other small things, I removed hard line breaks to make it consistent with the rest of the doc._

It has no bearing on the way the page is built, but I consider it good practice (it allows for saner parsing, and there's no need to reformat paragraphs after adding/removing text).

File: doc/reference/autoinstall-reference.rst (hunk 1 above)
Comment (dbungert) [inline doc/reference/autoinstall-reference.rst:22]: I'm good to merge this PR but I still want wording tweaks on this note, I'll follow-up with a PR for that.

**Review by dbungert — APPROVED** [review]

**Review by Chris-Peterson444 — APPROVED** [review]
> Yes, I noted it in the PR description:

Ahh. Sorry I missed that. But sounds good, I'll keep the formatting in mind for future work. Thanks!

**Comment by rkratky** [comment]
OK, merging now. @dbungert, ping me whenever ready to update the note.

**Pull request merged**